include/xeus/xhelper.hpp

```diff
@@ -0,0 +1,73 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay, Martin Renou          *
+* Copyright (c) 2016, QuantStack                                           *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XEUS_HELPER_HPP
+#define XEUS_HELPER_HPP
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "xeus/xeus.hpp"
+#include "xeus/xkernel.hpp"
+
+#include "nlohmann/json.hpp"
+#include "xtl/xvariant.hpp"
+
+namespace nl = nlohmann;
+
+namespace xeus
+{
+    XEUS_API void print_kernel_config(xkernel* kernel);
```

Comment:
- I think this method could be renamed in "print_starting_message", since it does a bit more than printing the config.
- Pointers should be avoided in public API, prefer references or const references
- This file is mainly for creating messages / dumping some outputs, so we may want to avoid pulling too many dependencies: `xeus/xkernel.hpp` depends on too many things that you don't want to depend on in the implementation of the kernel interpreter for instance. So accepting an `xkernel_fonciguration` parameter instead would help reducing the amount of pulled dependencies. It's not a problem to reapeat the call to `kernel->get_config();` in each kernel imo.